<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>692</width>
    <height>369</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>671</width>
      <height>321</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>2</number>
    </property>
    <widget class="QWidget" name="GronsfeldTab">
     <attribute name="title">
      <string>Gronsfeld</string>
     </attribute>
     <widget class="QTextEdit" name="m_gronsfeldDecodedMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>471</width>
        <height>81</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Decoded message:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Encoded message:</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_gronsfeldEncodedMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>471</width>
        <height>81</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="m_gronsfeldEncodeButton">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>120</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Encode</string>
      </property>
     </widget>
     <widget class="QPushButton" name="m_gronsfeldDecodButton">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>260</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Decode</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="m_gronsfeldCipherLine">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>50</y>
        <width>151</width>
        <height>22</height>
       </rect>
      </property>
      <property name="inputMethodHints">
       <set>Qt::ImhDigitsOnly|Qt::ImhFormattedNumbersOnly</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>30</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Cipher key:</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="XORTab">
     <attribute name="title">
      <string>XOR</string>
     </attribute>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Encoded message:</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_XOREncodedMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>471</width>
        <height>81</height>
       </rect>
      </property>
      <property name="locale">
       <locale language="Russian" country="RussianFederation"/>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Decoded message:</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_XORDecodedMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>471</width>
        <height>81</height>
       </rect>
      </property>
      <property name="locale">
       <locale language="Russian" country="RussianFederation"/>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>30</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Cipher key:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="m_XOREncodeButton">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>120</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Encode</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="m_XORCipherLine">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>50</y>
        <width>151</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="m_XORDecodButton">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>260</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Decode</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="RSATab">
     <attribute name="title">
      <string>RSA</string>
     </attribute>
     <widget class="QTextEdit" name="m_RSAMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>491</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_RSAEncoded">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>120</y>
        <width>491</width>
        <height>71</height>
       </rect>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_RSADecoded">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>210</y>
        <width>491</width>
        <height>71</height>
       </rect>
      </property>
      <property name="textInteractionFlags">
       <set>Qt::NoTextInteraction</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>4</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Message:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>100</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Encoded:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>190</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Decoded:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_10">
      <property name="geometry">
       <rect>
        <x>570</x>
        <y>20</y>
        <width>31</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Keys:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_11">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>60</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>P:</string>
      </property>
     </widget>
     <widget class="QLabel" name="m_pLine">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>60</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_12">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>90</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Q:</string>
      </property>
     </widget>
     <widget class="QLabel" name="m_qLine">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>90</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_17">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>120</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Public Key:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_18">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>140</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>E:</string>
      </property>
     </widget>
     <widget class="QLabel" name="m_eLine">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>140</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_19">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>170</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>N:</string>
      </property>
     </widget>
     <widget class="QLabel" name="m_nLine">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>170</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_20">
      <property name="geometry">
       <rect>
        <x>550</x>
        <y>200</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Private Key:</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_21">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>230</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>D:</string>
      </property>
     </widget>
     <widget class="QLabel" name="m_dLine">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>230</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_22">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>260</y>
        <width>16</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>N:</string>
      </property>
     </widget>
     <widget class="QLabel" name="m_nLine_2">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>260</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QPushButton" name="m_RSAEncode">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>100</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Encode</string>
      </property>
     </widget>
     <widget class="QPushButton" name="m_RSADecode">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>190</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Decode</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="ElGamalTab">
     <attribute name="title">
      <string>ElGamal</string>
     </attribute>
     <widget class="QLineEdit" name="m_elGamalPLine">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>30</y>
        <width>151</width>
        <height>22</height>
       </rect>
      </property>
      <property name="inputMethodHints">
       <set>Qt::ImhDigitsOnly</set>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_elGamalDecodedMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>471</width>
        <height>81</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="m_elGamalDecodButton">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>260</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Decode</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_13">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>150</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Encoded message:</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="m_elGamalEncodedMessage">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>170</y>
        <width>471</width>
        <height>81</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_14">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>10</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Decoded message:</string>
      </property>
     </widget>
     <widget class="QPushButton" name="m_elGamalEncodeButton">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>120</y>
        <width>99</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Encode</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_15">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>10</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>P:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="m_elGamalGLine">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>90</y>
        <width>151</width>
        <height>22</height>
       </rect>
      </property>
      <property name="inputMethodHints">
       <set>Qt::ImhDigitsOnly|Qt::ImhFormattedNumbersOnly</set>
      </property>
     </widget>
     <widget class="QLabel" name="label_16">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>70</y>
        <width>91</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>G:</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>692</width>
     <height>20</height>
    </rect>
   </property>
   <property name="inputMethodHints">
    <set>Qt::ImhDigitsOnly|Qt::ImhFormattedNumbersOnly</set>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
